cachi2/core/package_managers/yarn_classic/workspaces.py

```diff
@@ -0,0 +1,114 @@
+import json
+from contextlib import suppress
+from itertools import chain
+from pathlib import Path
+from typing import Any, Iterable
+
+import pydantic
+
+from cachi2.core.errors import PackageRejected
+from cachi2.core.models.input import YarnClassicPackageInput
+from cachi2.core.rooted_path import RootedPath
+
+
+class Workspace(pydantic.BaseModel):
```

Comment:
We should ensure that the `name` field is present in package.json for each workspace and fail otherwise